src/librustc_target/abi/call/wasm32.rs

```diff
@@ -1,16 +1,38 @@
-use crate::abi::call::{FnType, ArgType};
+use crate::abi::call::{FnType, ArgType, Uniform};
+use crate::abi::{HasDataLayout, LayoutOf, TyLayout, TyLayoutMethods};
 
-fn classify_ret_ty<Ty>(ret: &mut ArgType<'_, Ty>) {
-    ret.extend_integer_width_to(32);
```

Comment:
I think this should be kept.

Reply:
Can you explain why? There’s no need to expand i8s and i16s to i32s during the lowering to LLVM IR because LLVM will perform the necessary type legalization itself. In fact, it’s better not to do the expansion here so that legalization is handled by the same system for both C/C++ and Rust.

Reply:
I agree with @bjorn3, Clang appears to do this: https://github.com/llvm/llvm-project/blob/d01b4a786271c9ce9df43add56ec2e3fb40ce920/clang/lib/CodeGen/TargetInfo.cpp#L716

cc @rkruppe @nagisa @nikic